specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,313 @@
+
+@versioned(AzureCommunicationMessagesService.Versions)
+namespace AzureCommunicationMessagesService;
+
+@access(Access.public, "csharp")
+@doc("The type of message. Supports text, image, template.")
+enum CommunicationMessageType {
+  @doc("Text message type.")
+  text,
+
+  @doc("Image message type.")
+  image,
+
+  @doc("Template message type.")
+  template,
+}
+
+@doc("The template value kind.")
+enum MessageTemplateValueKind {
+  @doc("Message parameter type is text.")
+  text,
+
+  @doc("Message parameter type is image.")
+  image,
+
+  @doc("Message parameter type is document.")
+  document,
+
+  @doc("Message parameter type is video.")
+  video,
+
+  @doc("Message parameter type is location.")
+  location,
+
+  @doc("Message parameter type is quick action.")
+  quick_action,
+}
+
+@access(Access.public, "csharp")
+@doc("The WhatsApp button sub type.")
+enum WhatsAppMessageTemplateValueSubType {
+  @doc("The WhatsApp button sub type is quick reply.")
+  quickReply,
+
+  @doc("The WhatsApp button sub type is url.")
+  url,
+}
+
+@access(Access.public, "csharp")
+@doc("The Communication Messages Channel.")
+enum CommunicationMessagesChannel {
+  @doc("The Communication Messages Channel is WhatsApp.")
+  whatsApp,
+}
+
+@access(Access.public, "csharp")
+@doc("The aggregated template status.")
+enum MessageTemplateStatus {
+  @doc("Message template is approved.")
+  approved,
+
+  @doc("Message template is rejected.")
+  rejected,
+
+  @doc("Message template is pending.")
+  pending,
+
+  @doc("Message template is paused.")
+  paused,
+}
+
+@doc("Details of the message to send.")
+model SendNotificationRequest {
+  @doc("The Channel Registration ID for the Business Identifier.")
+  channelRegistrationId: uuid;
+
+  @doc("The native external platform user identifiers of the recipient.")
+  to: string[];
+
+  @doc("The type of message. Supports text, image, template.")
+  type: CommunicationMessageType;
+
+  @doc("Message content.")
+  content?: string;
+
+  @doc("""
+A media url for the file. Required if the type is one of the supported media
+types, e.g. image
+""")
+  mediaUri?: string;
+
+  @doc("The template object used to create templates.")
+  template?: MessageTemplate;
+}
+
+@doc("The template object used to create templates.")
+model MessageTemplate {
+  @doc("Name of the template.")
+  name: string;
+
+  @doc("The codes for the supported languages for templates.")
+  language: string;
+
+  @doc("The template values.")
+  values?: Record<MessageTemplateValue>;
+
+  @doc("The binding object to link values to the template specific locations")
+  bindings?: MessageTemplateBindings;
+}
+
+@doc("The class describes a parameter of a template.")
+model MessageTemplateValue {
+  @doc("The template value kind.")
+  kind: MessageTemplateValueKind;
+
+  @doc("The message template's text value information.")
+  text?: MessageTemplateValueText;
+
+  @doc("""
+The message template's media value information.
+Could be an image, document or
```

Comment:
This doc doesn't fit the `image` property. Let's remodel as discriminated types and then check the docs again.